### PR TITLE
feat: #147 Create new Way on UserPage

### PR DESCRIPTION
**Pull request by Ekaterina1994**

### CODE REVIEW

**Review by Howl404 — APPROVED** [review]

File: mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts
Comment (SergioAJS) [inline mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts:1]: Please check the placement of curly braces

File: mw-webapp/src/service/WayService.ts
Comment (SergioAJS) [inline mw-webapp/src/service/WayService.ts:1]: Please check the braces and make sure you have imported all the needed methods

File: mw-webapp/src/logic/userPage/UserPage.tsx [hunk 1]
```diff
@@ -17,6 +18,10 @@ export const UserPage = () => {
         level={HeadingLevel.h2}
         text="User page"
       />
+      <Button
+        value="Create new way"
+        onClick={() => WayPreviewDAL.createWayPreview(uuid!)}
```
Comment (SergioAJS) [inline mw-webapp/src/logic/userPage/UserPage.tsx:23]: Can you get rid of using the exclamation mark?

**Review by SergioAJS — CHANGES_REQUESTED** [review]

File: mw-webapp/src/logic/userPage/UserPage.tsx (hunk 1 above)
Comment (Ekaterina1994) [reply mw-webapp/src/logic/userPage/UserPage.tsx:23]: No I don't know how

Comment (Ekaterina1994) [reply mw-webapp/src/logic/userPage/UserPage.tsx]: Added interface for property from useParams but dont know is it right way

File: mw-webapp/src/service/WayService.ts
Comment (Ekaterina1994) [reply mw-webapp/src/service/WayService.ts:1]: Done

File: mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts
Comment (Ekaterina1994) [reply mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts:1]: Done

**Review by SergioAJS — APPROVED** [review]

File: mw-webapp/src/dataAccessLogic/GoalPreviewDAL.ts [hunk 2]
```diff
@@ -17,4 +18,22 @@ export class GoalPreviewDAL {
...
+  public static async createGoalPreview(): Promise<string> {
+    const goalPreviewWithoutUuid: GoalDTOWithoutUuid = {
+      studentUuid: "",
+      metricUuids: [""],
+      description: "",
+      estimationTime: 0,
+      timeUnit: TimeUnit.minute,
+    };
+
+    const newGoalUuid = await GoalService.createGoalDTO(goalPreviewWithoutUuid);
```
Comment (tritonJS826) [inline mw-webapp/src/dataAccessLogic/GoalPreviewDAL.ts:34]: newGoalUuid does not look like uuid, recheck please

File: mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts [hunk 3]
```diff
@@ -76,6 +76,23 @@ export class WayPreviewDAL {
     return waysPreview;
   }
 
+  /**
+   * Create WayPreview with empty fields and autogenerated uuid
+   */
+  public static async createWayPreview(userUuid: string) {
+    const newGoalUuid = await GoalPreviewDAL.createGoalPreview();
```
Comment (tritonJS826) [inline mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts:83]: Why does createGoalPreview returns  string with uuid?

File: mw-webapp/src/logic/userPage/UserPage.tsx [hunk 4]
```diff
@@ -1,22 +1,38 @@
...
 import {HeadingLevel, Title} from "src/component/title/Title";
+import {WayPreviewDAL} from "src/dataAccessLogic/WayPreviewDAL";
 import {OwnWaysTable} from "src/logic/waysTable/OwnWaysTable";
 import {pages} from "src/router/pages";
 import styles from "src/logic/userPage/UserPage.module.scss";
 
+/**
+ * Query param types
+ */
+interface QueryParamTypes extends Params {
```
Comment (tritonJS826) [inline mw-webapp/src/logic/userPage/UserPage.tsx:12]: It's better to keep url lprams for pages near pages

File: mw-webapp/src/logic/userPage/UserPage.tsx [hunk 5]
```diff
@@ -1,22 +1,38 @@
...
+   */
+  uuid: string;
+}
+
 /**
  * User page
  */
 export const UserPage = () => {
-  const {uuid} = useParams();
+  const {uuid} = useParams() as QueryParamTypes;
```
Comment (tritonJS826) [inline mw-webapp/src/logic/userPage/UserPage.tsx:24]: What will happened if uuid is undefined (handle this scenario)

**Review by tritonJS826 — APPROVED** [review]

File: mw-webapp/src/logic/userPage/UserPage.tsx (hunk 4 above)
Comment (Ekaterina1994) [reply mw-webapp/src/logic/userPage/UserPage.tsx:12]: Moved to separate file

File: mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts (hunk 3 above)
Comment (Ekaterina1994) [reply mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts:83]: Renamed

File: mw-webapp/src/dataAccessLogic/GoalPreviewDAL.ts (hunk 2 above)
Comment (Ekaterina1994) [reply mw-webapp/src/dataAccessLogic/GoalPreviewDAL.ts:34]: Renamed

File: mw-webapp/src/logic/userPage/UserPage.tsx (hunk 5 above)
Comment (Ekaterina1994) [reply mw-webapp/src/logic/userPage/UserPage.tsx:24]: If uuid is undefined  client will be navigate to page 404. This scenario works by code on  30 line. 
But I dont know is it right to make interface for uuid from useParams as string?

**Pull request merged**